test/src/unit-cppapi-enumerations.cc

```diff
@@ -0,0 +1,204 @@
+/**
+ * @file unit-cppapi-enumerations.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests the C++ API for enumeration related functions.
+ */
+
+#include <test/support/tdb_catch.h>
+#include "test/support/src/helpers.h"
+#include "tiledb/sm/cpp_api/tiledb"
+#include "tiledb/sm/cpp_api/tiledb_experimental"
+
+using namespace tiledb;
+
+struct TestData {
+  TestData()
+      : vfs_(ctx_) {
+    rm_array();
+  }
+
+  ~TestData() {
+    rm_array();
+  }
+
+  void rm_array() {
+    if (vfs_.is_dir(uri_)) {
+      vfs_.remove_dir(uri_);
+    }
+  }
+
+  Context ctx_;
+  VFS vfs_;
+  std::string uri_ = "enumeration_test_array";
+};
+
+void create_array(TestData& td);
+void write_array(
+    TestData& td, std::vector<int>& dim_vals, std::vector<int>& attr_vals);
+void read_array(
+    TestData& td,
+    QueryCondition* qc,
+    std::vector<int>& dim_vals,
+    std::vector<int>& attr_vals);
+
+TEST_CASE("C++ API: Enumeration creation fixed size", "[cppapi][enumeration]") {
+  TestData td;
+
+  std::vector<uint32_t> values = {1, 2, 3, 4, 5};
+  auto e = Enumeration::create(td.ctx_, values);
+
+  REQUIRE(e.type() == TILEDB_UINT32);
+  REQUIRE(e.cell_val_num() == 1);
+  REQUIRE(e.ordered() == false);
+
+  std::vector<uint32_t> data;
+  e.into_vector(data);
+  REQUIRE(data == values);
+}
+
+TEST_CASE("C++ API: Enumeration creation var sized", "[cppapi][enumeration]") {
+  TestData td;
+
+  std::vector<std::string> values = {"foo", "bar", "baz", "bingo", "bango"};
+  auto e = Enumeration::create(td.ctx_, values);
+
+  REQUIRE(e.type() == TILEDB_STRING_ASCII);
+  REQUIRE(e.cell_val_num() == std::numeric_limits<uint32_t>::max());
```

Comment:
Ah! For sure. I just through this in there because I didn't want to reference `constants::var_num` outside core. I'll do the same for all the others you noticed as well.

Reply:
Done.